modules/web-console/frontend/app/modules/agent/AgentManager.service.ts

```diff
@@ -891,6 +885,21 @@ export default class AgentManager {
         return this.clustersSecrets.get(clusterId).hasCredentials();
     }
 
+    /**
+     * Collect features supported by the cluster.
+     *
+     * @param cluster Cluster.
+     * @return all supported features.
+     */
+    allFeatures(cluster: AgentTypes.ClusterStats): AgentTypes.IgniteFeatures[] {
+        return _.reduce(AgentTypes.IgniteFeatures, (acc, featureId) => {
```

Comment:
The new API looks awkward, but it's better than before. We should probably iterate on this later.

Reply:
I tried to investigate this and found that I could not define an object with keys as in enumeration keys, only as values.

```
enum A {
  A1 = 10
}

type B = {
  [x in A]: boolean
}

const features: B = {
  10: true
}
```